modules/flowable-cmmn-engine/src/test/java/org/flowable/cmmn/test/tenant/ChangeTenantIdCaseInstanceTest.java

```diff
@@ -0,0 +1,264 @@
+/* Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.flowable.cmmn.test.tenant;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.Set;
+
+import org.flowable.cmmn.api.runtime.CaseInstance;
+import org.flowable.cmmn.api.runtime.CaseInstanceBuilder;
+import org.flowable.cmmn.engine.test.FlowableCmmnTestCase;
+import org.flowable.common.engine.api.tenant.ChangeTenantIdResult;
+import org.flowable.task.api.Task;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.internal.util.collections.Sets;
+
+public class ChangeTenantIdCaseInstanceTest extends FlowableCmmnTestCase {
+
+    private static final String TEST_TENANT_A = "test-tenant-a";
+    private static final String TEST_TENANT_B = "test-tenant-b";
+    private static final String TEST_TENANT_C = "test-tenant-c";
+    
+    private String deploymentIdWithTenantA;
+    private String deploymentIdWithTenantB;
+    private String deploymentIdWithTenantC;
+    private String deploymentIdWithoutTenant;
+
+    @Before
+    public void setUp() {
+        this.deploymentIdWithTenantA = cmmnRepositoryService.createDeployment()
+                .addClasspathResource("org/flowable/cmmn/test/tenant/caseWithMilestone.cmmn").tenantId(TEST_TENANT_A)
+                .deploy().getId();
+        addDeploymentForAutoCleanup(deploymentIdWithTenantA);
+        this.deploymentIdWithTenantB = cmmnRepositoryService.createDeployment()
+                .addClasspathResource("org/flowable/cmmn/test/tenant/caseWithMilestone.cmmn").tenantId(TEST_TENANT_B)
+                .deploy().getId();
+        addDeploymentForAutoCleanup(deploymentIdWithTenantB);
+        this.deploymentIdWithTenantC = cmmnRepositoryService.createDeployment()
+                .addClasspathResource("org/flowable/cmmn/test/tenant/caseWithMilestone.cmmn").tenantId(TEST_TENANT_C)
+                .deploy().getId();
+        addDeploymentForAutoCleanup(deploymentIdWithTenantC);
+        this.deploymentIdWithoutTenant = cmmnRepositoryService.createDeployment()
+                .addClasspathResource("org/flowable/cmmn/test/tenant/caseWithMilestoneDup.cmmn").deploy().getId();
+        addDeploymentForAutoCleanup(deploymentIdWithoutTenant);
+    }
+
+    @Test
+    public void testDeployments() {
```

Comment:
Each test needs to be self explanatory. There are 2 tests here, where one test tests the setup for the other test. Let's move those 2 things together.

Reply:
Done